decidim-core/lib/decidim/resourceable.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+require "active_support/concern"
+
+module Decidim
+  # A concern with the features needed when you want a model to be able to create
+  # links from it to another resource.
+  module Resourceable
+    extend ActiveSupport::Concern
+
+    included do
+      # An association with all the links that point to this model.
+      has_many :resource_links_to, as: :to, class_name: "Decidim::ResourceLink"
+
+      # An association with all the links that are originated from this model.
+      has_many :resource_links_from, as: :from, class_name: "Decidim::ResourceLink"
+
+      # Finds all the linked resources to or from this model for a given resource
+      # name and link name.
+      #
+      # resource_name - The String name of the resource manifest exposed by a feature.
+      # link_name     - The String name of the link between this model and the target resource.
+      #
+      # Returns an ActiveRecord::Relation.
+      def linked_resources(resource_name, link_name)
+        scope = sibling_scope(resource_name)
+
+        from = scope
+               .joins(:resource_links_from)
+               .where(decidim_resource_links: { name: link_name.to_s, to_id: id, to_type: self.class.name })
+
+        to = scope
+             .joins(:resource_links_to)
+             .where(decidim_resource_links: { name: link_name.to_s, from_id: id, from_type: self.class.name })
+
+        scope.where(id: from).or(scope.where(id: to)).distinct
+      end
+
+      # Builds an ActiveRecord::Relation in order to load all the resources
+      # that are in the same parent as this model.
+      #
+      # resource_name - The String name of the resource manifest exposed by a feature.
+      #
+      # Returns an ActiveRecord::Relation.
+      def sibling_scope(resource_name)
```

Comment:
Not really, I'm using it at `CloseMeetingForm` to load all the proposals in the same process.